src/ActionMenu.tsx

```diff
@@ -123,8 +124,13 @@ const Overlay: React.FC<MenuOverlayProps> = ({children, align = 'start', ...over
       onOpen={openWithFocus}
       onClose={onClose}
       align={align}
-      overlayProps={overlayProps}
+      overlayProps={{
+        usePortal: false, // render inline for natural focus management
+        sx: merge({zIndex: 1}, propsSx as SxProp), // render above overlay contents (story/internal-components-overlay--memex-nested-overlays)
+        ...overlayProps
+      }}
       focusZoneSettings={{focusOutBehavior: 'wrap'}}
+      focusTrapSettings={{disabled: true}}
```

Comment:
Menu should not have focus trap: Pressing Tab (or Shift + Tab) should close the menu.

(The keyboard navigation with arrow keys is handled by focusZone, not focusTrap)